sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -982,9 +989,13 @@ public WriteResult expand(PCollection<T> input) {
                 + " PCollection.");
         StreamingInserts<DestinationT> streamingInserts =
             new StreamingInserts<>(getCreateDisposition(), dynamicDestinations);
+        streamingInserts.setInsertRetryPolicy(getFailedInsertRetryPolicy());
         streamingInserts.setTestServices(getBigQueryServices());
         return rowsWithDestination.apply(streamingInserts);
       } else {
+        checkArgument(getFailedInsertRetryPolicy() == null,
+            "Record-insert retry policies are not supported when using BigQuery load jobs.");
```

Comment:
I didn't want to write that, because tying "unbounded collection" to BQ streaming inserts is an artifact of the current code, and arguably not a correct one. There are plans to fix this and have the insert method chosen independently of the input PCollection.